client/control/actions/FilterActions.js

```diff
@@ -0,0 +1,113 @@
+import {
+  makeApiRequest,
+  recipesNeedFetch,
+  recipesReceived,
+  filtersReceived,
+} from 'control/actions/ControlActions';
+
+import {
+  getFilterParamString,
+} from 'control/selectors/FiltersSelector';
+
+const LOAD_FILTERS = 'LOAD_FILTERS';
+const SET_FILTER = 'SET_FILTER';
+const SET_TEXT_FILTER = 'SET_TEXT_FILTER';
+const SET_ALL_FILTERS = 'SET_ALL_FILTERS';
+
+
+/**
+ * Load list of possible filters from remote API.
+ * This is stored in the `filters` reducer, and
+ * later used to populate relevant RecipeFilters components.
+ */
+function loadFilters() {
+  return dispatch =>
+    dispatch(makeApiRequest('fetchFilters'))
+      .then(recipes => dispatch(filtersReceived(recipes)));
+}
+
+/**
+ * Given an option and its parent group, update the
+ * filter state based on the `isEnabled` prop
+ *
+ * @param  {Object}  group     Group the filter belongs to
+ * @param  {Object}  option    Option that was affected
+ * @param  {Boolean} isEnabled Is the option selected?
+ */
+function selectFilter({ group, option, isEnabled }) {
+  return dispatch => {
+    dispatch({
```

Comment:
Does this need to be a thunked action creator? It looks like this could just return the action directly, instead of the return dispatch => { ... } dance.